broker/httpd/instrumentedWriter.go

```diff
@@ -0,0 +1,232 @@
+package httpd
+
+/* Original Instpiration comes from https://github.com/mash/go-accesslog/
+
+ */
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type LogRecord struct {
+	Time                                      time.Time
+	Ip, Method, Uri, Protocol, Username, Host string
+	Status                                    int
+	Size                                      int64
+	ElapsedTime                               time.Duration
+	RequestHeader                             http.Header
+	CustomRecords                             map[string]string
+}
+
+type LoggingWriter struct {
+	http.ResponseWriter
+	logRecord LogRecord
+}
+
+var (
+	rpcDurations = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "service_http_request_processing_duration_seconds",
+			Help:       "RPC latency distributions.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"code"},
```

Comment:
Please add tricorder metrics.

Reply:
There is no simple way to add all potential codes... not worth it.